crml/erc20-peg/src/lib.rs

```diff
@@ -125,6 +152,57 @@ decl_module! {
 
 		fn deposit_event() = default;
 
+		/// Check and process outstanding claims
+		fn on_initialize(now: T::BlockNumber) -> Weight {
+			let mut weight: u64 = 10_000_000;
+			if ClaimSchedule::<T>::contains_key(now) {
+				match Self::ready_blocks() {
+					Some(mut blocks) => {
+						// TODO Check that ready_blocks doesn't overflow
+						blocks.push(now);
+						ReadyBlocks::<T>::put(blocks);
+					},
+					None => ReadyBlocks::<T>::put(vec![now]),
+				}
+				weight += 10_000_000;
+			}
+			weight as Weight
+		}
+
+		/// Check and process outstanding claims
+		fn on_idle(now: T::BlockNumber, remaining_weight: Weight) -> Weight {
+			let initial_read_cost = 10_000_000;
+			// Ensure we have enough weight to perform the initial read
+			if remaining_weight <= initial_read_cost {
+				return 0;
+			}
+			// Check if there are any blocks pending, else return early
+			let ready_blocks = Self::ready_blocks();
+			if !ready_blocks.is_some() {
+				return initial_read_cost;
+			}
+			let mut ready_blocks: Vec<T::BlockNumber> = ready_blocks.unwrap();
+			// Process as many claims as we can
+			let weight_each: Weight = 50_000_000;
+			let max_claims = (remaining_weight / weight_each).saturated_into::<u8>();
+
+			let mut processed_claims: u8 = 0;
+
+			if ready_blocks[0] <= now {
+				while processed_claims < max_claims && !ready_blocks.is_empty() {
+					if ClaimSchedule::<T>::contains_key(ready_blocks[0]) {
+						processed_claims += Self::process_claims_at(ready_blocks[0], max_claims - processed_claims);
+					} else {
+						ready_blocks.remove(0);
+					}
```

Comment:
try to avoid checking storage every loop
`.remove(0)` can be expensive since it shifts all elements left

something like `put(ready_blocks[completed_count:]` may be nicer